src/pages/index.js

```diff
@@ -1,120 +1,267 @@
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
 import Section from "../components/Section.js";
-import { initialCards, selectors, config } from "../utils/constants.js";
+import UserInfo from "../components/UserInfo.js";
+import { selectors, config } from "../utils/constants.js";
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "0d34a92c-1da7-4adc-abc0-e2b3b38a4b1f",
+    "Content-Type": "application/json",
+  },
+});
 
-// Find Profile and Add Card buttons.
-const profileEditButton = document.querySelector("#profile-edit-button");
+// Find buttons.
+const editAvatButton = document.querySelector("#profile-avatar-container");
+const editProfButton = document.querySelector("#profile-edit-button");
 const addCardButton = document.querySelector("#add-card-button");
 
 // Find edit form input elements.
-const profileTitleInput = document.querySelector("#profile-title-input");
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
+const profNameInput = document.querySelector("#profile-title-input");
+const profAboutInput = document.querySelector("#profile-about-input");
 
 //  Find form elements.
-const profileEditForm = document.forms["edit-card-form"];
-const addCardModalForm = document.forms["add-card-form"];
+const profEditForm = document.forms["edit-card-form"];
+const addCardForm = document.forms["add-card-form"];
+const avatEditForm = document.forms["avatar-edit-form"];
 
 /* ------------------- Form Validation -------------------- */
 
-const addFormValidator = new FormValidator(config, addCardModalForm);
-const editFormValidator = new FormValidator(config, profileEditForm);
+const addFormValidator = new FormValidator(config, addCardForm);
+const editFormValidator = new FormValidator(config, profEditForm);
+const avatEditFormValidator = new FormValidator(config, avatEditForm);
 
 /* --------------------- Card Handler --------------------- */
 
-// Create image Popup instance.
-const cardPreviewPopup = new PopupWithImage(selectors.previewPopup);
+let cardSection;
+let card;
 
-// This function creates a new card
-function createCard(cardData) {
-  const cardElement = new Card(
-    {
-      cardData,
-      handleImageClick: (imageData) => {
-        cardPreviewPopup.open(imageData);
-      },
-    },
-    selectors.cardTemplate
-  );
+/* --------------- Edit Avatar Card --------------- */
 
-  return cardElement.getView();
+// Edit profile Avatar thumbnail imageData {avatURL}
+const editAvatPopup = new PopupWithForm(
+  selectors.editAvatFormModal,
+  handleAvatarSubmit,
+  "Save",
+  "Saving..."
+);
+
+function handleAvatarSubmit(imageData) {
+  editAvatPopup.showLoading();
+  api
+    .setAvatarAPI(imageData.avatURL)
+    .then((res) => {
+      userInfo.setAvatar(imageData.avatURL);
+      editAvatPopup.close();
+    })
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**